src/newViews/Tagging/Sources/LeadEditModal/LeadEditForm/index.tsx

```diff
@@ -0,0 +1,243 @@
+import React, { useState, useMemo } from 'react';
+import {
+    _cs,
+    unique,
+    isTruthyString,
+    compareNumber,
+    isDefined,
+} from '@togglecorp/fujs';
+import {
+    PendingMessage,
+    TextInput,
+    DateInput,
+    SegmentInput,
+    SelectInput,
+    Checkbox,
+    TextArea,
+} from '@the-deep/deep-ui';
+import {
+    EntriesAsList,
+    Error,
+    getErrorObject,
+    getErrorString,
+} from '@togglecorp/toggle-form';
+
+import { useRequest } from '#utils/request';
+import OrganizationSelectInput from '#newComponents/input/OrganizationSelectInput';
+import OrganizationMultiSelectInput from '#newComponents/input/OrganizationMultiSelectInput';
+
+import { BasicOrganization } from '#typings';
+import {
+    PartialFormType,
+    LeadOptions,
+    Priority,
+} from './schema';
+import ConfidentialityInput from './ConfidentialityInput';
+import EmmStats from './EmmStats';
+
+import styles from './styles.scss';
+
+// FIXME: Use translations throughout the page
+
+const idSelector = (item: { id: number }) => item.id;
+const titleSelector = (item: { title: string}) => item.title;
+const displayNameSelector = (item: { displayName: string }) => item.displayName;
+
+const keySelector = (item: Priority) => item.key;
+const valueSelector = (item: Priority) => item.value;
+
+interface Props {
+    className?: string;
+    setFieldValue: (...values: EntriesAsList<PartialFormType>) => void;
+    value: PartialFormType;
+    error: Error<PartialFormType> | undefined;
+    initialValue: PartialFormType;
+}
+
+function LeadEditForm(props: Props) {
+    const {
+        className,
+        value,
+        initialValue,
+        error: riskyError,
+        setFieldValue,
+    } = props;
+    const error = getErrorObject(riskyError);
+
+    const optionsRequestBody = useMemo(() => ({
+        projects: [initialValue.project],
+        leadGroups: [], // this will not fetch any leadGroups
+        organizations: unique(
+            [
+                initialValue.source,
+                ...(initialValue.authors || []),
+            ].filter(isDefined),
+            id => id,
+        ),
+    }), [initialValue]);
+
+    const [
+        sourceOrganizationOptions,
+        setSourceOrganizationOptions,
+    ] = useState<BasicOrganization[] | undefined | null>();
+
+    const [
+        authorOrganizationOptions,
+        setAuthorOrganizationOptions,
+    ] = useState<BasicOrganization[] | undefined | null>();
+
+    const {
+        pending,
+        response: leadOptions,
+    } = useRequest<LeadOptions>({
+        method: 'POST',
+        url: 'server://lead-options/',
+        body: optionsRequestBody,
+        failureHeader: 'Lead Options',
```

Comment:
shouldn't we refresh the source table on every added source?

Reply:
Yes we should, but now its a bit difficult. Need to discuss this. I've added a todo for now.